### PR TITLE
Floating messages comissioned by MI2 coded by Julia Longtin

### PR DESCRIPTION
**Pull request by teryhill**
Floating messages comissioned by MI2 coded by Julia Longtin #267

### CODE REVIEW

File: interface/main/messages/hoverover.js.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            ;
+        }
+        ;
+    }
+    ;
+};
+$(function () {
+    $('<iframe style="width:0px;height:0px;border:0px;" id="hiddenIFrame" onload="hiddenIFrameload()" src="<?php echo $GLOBALS['webroot']; ?>/interface/main/messages/message_status.php"/>').appendTo('body');
+    $('#hiddenIFrame').attr('src', "<?php echo $GLOBALS['webroot']; ?>/interface/main/messages/message_status.php?skip_timeout_reset=1");
+});
```
Comment (aethelwulffe) [inline interface/main/messages/hoverover.js.php]: EOF newline

File: interface/main/messages/hoverover.js.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+// TODO - make this a Global config for the interval  (Time between displays was 30000)
+setInterval(function () {
+    $('#hiddenIFrame').attr('src', "<?php echo $GLOBALS['webroot']; ?>/interface/main/messages/message_status.php?skip_timeout_reset=1");
+}, parsetime);
+
+iframeloaded = function () {
+    if ($('#hiddenIFrame').contents().find('#notices').text()) {
+        var noticecount = $('#hiddenIFrame').contents().find('#noticecount').text();
+        if (noticecount > 0) {
+            for (var i = 0; i < noticecount; i++) {
```
Comment (aethelwulffe) [inline interface/main/messages/hoverover.js.php]: This for loop right?

File: interface/main/messages/message_status.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    /* uncomment this for demos
+    echo '<div id="notice'.$notices.'"><div class="sticky">1</div><div class="UUID">VMTESTING1</div><div class="colour">red</div><div class="title">'.xlt('WARNING').'</div><div class="text">'.xlt('This VM is for TESTING ONLY').'</div>';
+    $notices++;
+    */
+    if ($notices > 0) {
+        echo '<div id="noticecount">' . $notices . '</div>';
+    }
+    ?></div>
+</body>
+</html>
```
Comment (aethelwulffe) [inline interface/main/messages/message_status.php]: EOF newline

File: library/globals.inc.php [hunk 4]
```diff
@@ -848,6 +853,32 @@
       xl('The default code type to search for in the Fee Sheet.')
     ),
     
+    'floating_message_alerts' => array(
```
Comment (aethelwulffe) [inline library/globals.inc.php]: adding a timer interval selection for globals would be nice.

Comment (teryhill) [inline library/globals.inc.php]: it has a 20 30 40 or 50 sec refresh see line 870

File: library/globals.inc.php [hunk 5]
```diff
@@ -115,7 +115,12 @@
                                'ptkr_visit_reason',
                                'checkout_roll_off',
                                'ptkr_pt_list_new_window',                               
-                               'erx_import_status_message');
+                               'erx_import_status_message',
+                               'floating_message_alerts',
+                               'floating_message_alerts_timer',
+                               'floating_message_alerts_allergies',
+                               'cms_top_margin_default',
```
Comment (tmccormi) [inline library/globals.inc.php]: cms stuff should not part be of this commit

Comment (teryhill) [inline library/globals.inc.php]: Removed

**Pull request merged**